### PR TITLE
M3-3681 Improve Graphs UX

### PR DESCRIPTION
**Pull request by alioso**
## M3-3681 Improve Graphs UX

- if we use transparency on any bg color we get wrong superposition hex values
- If we stack (or if we don’t) we need the order right to get to display the right things

By adding our own legend we lost an important feature of GraphJS, which is clicking on the legend to toggle the graphs. This PR adds this, remove the `stacked` option so now some lines may be hidden behind others, but with the option to toggle people may now see very small data on a relative scale, which is great.

I duplicated the styles from the MetricStyles in order to keep the legacy legend as this is more of a POC, but hopefully we can get rid of `MetricDisplay` if this is a go. I only replaced them in `LinodeSummary`

note: on `LineGraph` implementation, `data` and `legendRows` props must have the same data order in order to display in the right row.

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/components/LineGraph/LineGraph.tsx [hunk 1]
```diff
@@ -24,16 +34,21 @@ export interface Props {
   suggestedMax?: number;
   data: DataSet[];
   timezone: string;
+  rowHeaders?: Array<string>;
+  legendRows?: Array<any>;
```
Comment (Jskobos) [inline packages/manager/src/components/LineGraph/LineGraph.tsx]: This needs to be typed

**Comment by martinmckenna** [comment]
FWIW, I don't know if removing the patch was the right move.

Here's a screenshot I took on develop from 9:20AM EST time (but my timezone in Cloud is GMT):

Screen Shot 2019-12-12 at 9 25 00 AM


And here's a screenshot on this branch from 9:22AM EST (but my timezone in Cloud is GMT):

Screen Shot 2019-12-12 at 9 27 06 AM

So the issue here is that the second screenshot is not observing my timezone because it's not set to New_York, but still telling me the last stat in east coast time.

**Comment by alioso** [comment]
@Jskobos @martinmckenna reverted the bump version and re-added the patch for now to avoid introducing regressions to other charts and keep the time display timezone-accurate.

Also improved typing

**Comment by martinmckenna** [comment]
Still getting some regressions from the x axis when `showToday` is `true`:

Here's develop branch:

Screen Shot 2019-12-13 at 9 55 15 AM

And then this PR:

Screen Shot 2019-12-13 at 9 55 29 AM

**Comment by johnwcallahan** [comment]
Looking good. The selectable legends work well. 

I noticed a regression when selecting "Last 30 days" (or per-month) options on Linode Summary:

Screen Shot 2019-12-13 at 10 10 51 AM

**Pull request merged**